src/util/string_utils.js

```diff
@@ -10,6 +10,11 @@ const email_regexp = /^((([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\
 const escape_regexp = /[\\^$.*+?()[\]{}|]/g;
 const access_key_regexp = /^[a-zA-Z0-9]{20}$/;
 const secret_key_regexp = /^[a-zA-Z0-9+/]{40}$/;
+// regex for IAM service
+const AWS_IAM_PATH_REGEXP = /(\u002F)|(\u002F[\u0021-\u007E]+\u002F)/;
+const AWS_USERNAME_REGEXP = /[\w+=,.@-]+/;
+const AWS_LIST_MARKER = /[\u0020-\u00FF]+/;
+const AWS_ACCESS_KEY_ID = /[\w]+/;
```

Comment:
is \w really matching the correct character set for access keys?

Reply:
I took the pattern from the docs, for example ([UpdateAccessKey](https://docs.aws.amazon.com/IAM/latest/APIReference/API_UpdateAccessKey.html#API_UpdateAccessKey_RequestParameters)):

> Pattern: [\w]+

Explanation of the pattern:
> a string of characters that can consist of any upper or lowercased letter or digit.

Reply:
in javascript \w includes also underscore

Reply:
and we already have an access key regex defined - 
https://github.com/noobaa/noobaa-core/blob/7b0d3e3ccbcbf45365877ef619e52fa0bac5626e/src/util/string_utils.js#L11-L12

Reply:
1. Why underscore should be included?
I did a test in AWS IAM - I sent access key with _ and received an error `InvalidClientTokenId`:

> The security token included in the request is invalid

and then I sent with # and received an error `ValidationError`:

> The specified value for accessKeyId is invalid. It must contain only alphanumeric characters

Note: I used `aws iam get-access-key-last-used --access-key-id <>`. took a valid access key id and changed one of its characters.

2. It is not the same regex - for example, it sets the length to be exactly 20 while for an input it can be:

> Length Constraints: Minimum length of 16. Maximum length of 128.

Reply:
so how about we keep the pattern as is but to avoid confusion with other definitions we rename this const to something like AWS_IAM_ACCESS_KEY_INPUT_REGEXP ?

Reply:
I will change it.
I still think this name will be confusing because later we might use this validation in the CLI.

Reply:
but you have to agree that having two different regexp that are names aws_access_key is way more confusing

Reply:
I agree with this, I comment on it in case we have a better name.